qemu/qmp.go

```diff
@@ -639,6 +640,9 @@ func (q *QMP) ExecuteDeviceAdd(ctx context.Context, blockdevID, devID, driver, b
 	if bus != "" {
 		args["bus"] = bus
 	}
+	if addr != "" {
```

Comment:
I'm not sure about this.  The name of the device address property is bus specific.  "addr" as used here is PCI specific.  As we can specify different buses it doesn't feel right to hardcode a PCI specific key name in here.  I guess the best thing to do would be to only set the address if the bus is specified and to use the bus type to figure out what key to use.  @mcastelino any thoughts on this?

Reply:
@devimc can we generate the address based on PCI properties instead of using a string. That may help.

Reply:
Here's the document that concerned me

https://github.com/qemu/qemu/blob/master/docs/qdev-device-use.txt#L9

Reply:
unfortunately in Q35 we have to specify the address (number between 0x0 - 0x1f) hence each device must have a unique address, probably the best way to address this conflict could be adding a new method to hot plug just PCI devices ```ExecutePCIDeviceAdd```

Reply:
```bus``` must be specified too, because we want to hot plug devices on pci(e) bridges

Reply:
Fair enough.